### DISCUSSION TITLE
Dynamic solar charging for small (single phase) PV systems

### DISCUSSION THREAD

**Discussion by adjego** [dingo35/SmartEVSE-3.5#78] [Ideas]
I was having an idea to use the most PV as possible on a day for charging.

Now when you set solar_start_current to 6A and solar_max_import to 0 it will only start when there is 6A available for returning to the grid. This works as expected and is fine.

My idea on top of this would be the following:

- Set the maximum solar_max_import but also have that dynamically change. So setting it to for example max 6A, but when there is enough from solar alone, it will not use grid power anymore.
- Set solar_start_current to 1A and solar_max_import to 5A, and have that dynamically change to solar_start_current -> 6A and solar_max_import -> 0. So in the morning it will already charge when there is solar, and it will stop using additional grid power when there is enough solar generation.

**Comment by zulufoxtrot** [comment]
I think I have the same need as you. I opened another discussion, can you confirm it is what you want?
https://github.com/dingo35/SmartEVSE-3.5/discussions/160
